src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -198,6 +198,15 @@ default Optional<Result> findFirstByParticipationIdAndRatedWithSubmissionOrderBy
             """)
     Optional<Result> findByIdWithEagerFeedbacks(@Param("resultId") long resultId);
 
+    @Query("""
+            SELECT r
+            FROM Result r
+                LEFT JOIN FETCH r.feedbacks f
+                LEFT JOIN FETCH f.testCase
+            WHERE r.id IN :resultIds
+            """)
+    List<Result> findAllByIdWithEagerFeedbacks(@Param("resultIds") List<Long> resultIds);
```

Comment:
Is it necessary to talk of `feedbacks` instead of `feedback` for some reason? Feedback is a mass noun that cannot be pluralized. If there is no technical reason to use the `feedbacks` please stick to feedback.